packages/nvidia-k8s-device-plugin/0001-Add-CDI-specs-for-the-all-device.patch

```diff
@@ -0,0 +1,45 @@
+From 1ac450cfc861200a77685dd0ba77394963217502 Mon Sep 17 00:00:00 2001
+From: Jingwei Wang <jweiw@amazon.com>
+Date: Tue, 15 Apr 2025 18:17:21 +0000
+Subject: [PATCH]  Add CDI specs for the "all" device
+
+The Device Plugin cannot generate CDI specifications
+when NVIDIA_VISIBLE_DEVICES is set to "all". This limitation breaks compatibility
+with containers that depend on this setting, e.g. DCGM Exporter.
+
+Fix this by mimicking what nvidia-ctk does[1], and add the "all" device
+to the CDI specifications.
+
+[1]: https://github.com/NVIDIA/nvidia-container-toolkit/blob/main/cmd/nvidia-ctk/cdi/generate/generate.go#L300-L303
+
+Signed-off-by: Jingwei Wang <jweiw@amazon.com>
+---
+ internal/cdi/cdi.go | 5 +++++
+ 1 file changed, 5 insertions(+)
+
+diff --git a/internal/cdi/cdi.go b/internal/cdi/cdi.go
+index fae421b..0c5a4b4 100644
+--- a/internal/cdi/cdi.go
++++ b/internal/cdi/cdi.go
+@@ -38,6 +38,7 @@ import (
+ 
+ const (
+ 	cdiRoot = "/var/run/cdi"
++  allDeviceName = "all"
```

Comment:
You should fix the indentation here, so that we could contribute this upstream if required